gql/lib/src/schema/definitions/base_types.dart

```diff
@@ -0,0 +1,264 @@
+/// ignore_for_file: prefer_constructors_over_static_methods
+/// Contents:
+/// * [GraphQLEntity]
+/// * [AbstractType]
+/// * [NamedType]
+/// * [ListType]
+/// * [Argument]
+/// * [Directive]
+/// * [TypeSystemDefinition]
+/// * [TypeDefinition]
+part of "definitions.dart";
+
+@immutable
+abstract class GraphQLEntity extends Equatable {
```

Comment:
No need to extends `Equatable`.

You can use `DeepCollectionEquality` to implement `operator==` and `hashCode`. See `Node` in `lib/src/ast/ast.dart`.

After that, the dependency on `package:equatable` will be redundant.